### PR TITLE
[FLINK-5257] [table] display optimized logical plan when explaining table

### PR DESCRIPTION
**Pull request by KurtYoung**

### CODE REVIEW

**Comment by fhueske** [comment]
Thanks for the PR @KurtYoung.
+1 to merge

**Comment by fhueske** [comment]
merging

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/BatchTableEnvironment.scala [hunk 1]
```diff
@@ -275,17 +278,27 @@ abstract class BatchTableEnvironment(
...
 
+  /**
+    * Translates a logical [[RelNode]] into a [[DataSet]].
+    *
+    * @param logicalPlan The root node of the relational expression tree.
+    * @param tpe         The [[TypeInformation]] of the resulting [[DataSet]].
+    * @tparam A The type of the resulting [[DataSet]].
+    * @return The [[DataSet]] that corresponds to the translated [[Table]].
+    */
+  protected def translate[A](logicalPlan: RelNode)(implicit tpe: TypeInformation[A]): DataSet[A] = {
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/BatchTableEnvironment.scala]: Just noticed when checking the PR for the merge. This line exceeds the 100 char limit of our Scala style.
Please run a local build before opening a PR to catch these issues. Usually it is sufficient to build the modules you touched in the PR (`flink-table` in this case) with `mvn clean install`.

Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/BatchTableEnvironment.scala]: I'll fix that before merging. No need to update the PR.

Comment (KurtYoung) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/BatchTableEnvironment.scala]: Sorry for the inconvenience, I will pay attention to this kind of issues next time.

**Pull request closed without merging**